File: src/main/java/io/gaegul/ch01/DecoyDuck.java
package io.gaegul.ch01;

import io.gaegul.ch01.behavior.FlyNoWay;
import io.gaegul.ch01.behavior.MuteQuack;

public class DecoyDuck extends Duck {

	public DecoyDuck() {
		super(new FlyNoWay(), new MuteQuack());
	}

	@Override
	public void display() {
		System.out.println("가짜 오리");
	}
}

File: src/main/kotlin/io/gaegul/ch01/DecoyDuck.kt
package io.gaegul.ch01

import io.gaegul.ch01.behavior.FlyNoWay
import io.gaegul.ch01.behavior.MuteQuack

class DecoyDuck: Duck(
    MuteQuack(), FlyNoWay()
) {
    override fun display() {
        println("가짜 오리")
    }
}